pytorch_lightning/loops/epoch/prediction_epoch_loop.py

```diff
@@ -68,8 +68,8 @@ def on_run_start(  # type: ignore[override]
         void(dataloader_iter, dataloader_idx)
         self._dl_max_batches = dl_max_batches
         self._num_dataloaders = num_dataloaders
-        self._seen_batch_indices = self._get_batch_indices(dataloader_idx)
         self.return_predictions = return_predictions
+        self._seen_batch_indices = self._get_batch_indices(dataloader_idx)
```

Comment:
Good catch, let's add a comment

```suggestion
        # this call requires that `self.return_predictions` is set
        self._seen_batch_indices = self._get_batch_indices(dataloader_idx)
```